pkg/controller/localvolumeset/finalizer.go

```diff
@@ -0,0 +1,60 @@
+package localvolumeset
+
+import (
+	"context"
+	"fmt"
+
+	localv1alpha1 "github.com/openshift/local-storage-operator/pkg/apis/local/v1alpha1"
+	"github.com/openshift/local-storage-operator/pkg/common"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *LocalVolumeSetReconciler) syncFinalizer(request reconcile.Request) error {
+	lvSet := &localv1alpha1.LocalVolumeSet{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, lvSet)
```

Comment:
Doesn't this function gets called after fetching localvolumeset object? Do we need to fetch it again?

Reply:
The client reads from a cache, so it's not making a separate API call

Reply:
Where is this cache you speak of?

Reply:
[Here](https://github.com/kubernetes-sigs/controller-runtime/blob/dea6dee6c3e1ae2a2c0555f37ab28266105a1206/pkg/client/split.go#L115), but I'll change it because of uncertainty about when the cache is invalidated and there's not need to check if it's cached as we already have the value.